### PR TITLE
Use adapter.js

### PR DESCRIPTION
**Pull request by lodoyun**
**Description**

This PR updates most of the webrtc API calls and  adds [adapter.js](https://github.com/webrtc/adapter) to simplify supporting different browser.
* There's a new stack called `baseStack` that is inherited by both `ChromeStableStack` and `FirefoxStack` and takes care of most of the functionality. Only small differences are now implemented in each browsers' stack.
* Added helper methods for SDP mangling
* Updated Connection getUserMedia API.
* Removed bowserStack as it was largely unsupported.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

No changes to the API

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: erizo_controller/erizoClient/src/Stream.js [hunk 1]
```diff
@@ -167,7 +172,7 @@ Erizo.Stream = (specInput) => {
     const options = optionsInput || {};
     that.elementID = elementID;
     let player;
-    if (that.hasVideo() || this.hasScreen()) {
+    if (that.hasVideo() || that.hasScreen()) {
```
Comment (jcague) [inline erizo_controller/erizoClient/src/Stream.js]: good catch!

File: erizo_controller/erizoClient/src/webrtc-stacks/ChromeStableStack.js [hunk 2]
```diff
@@ -1,66 +1,13 @@
...
-  const that = {};
-  const WebkitRTCPeerConnection = webkitRTCPeerConnection;
-  const defaultSimulcastSpatialLayers = 2;
+  L.Logger.info('Starting Chrome stable stack', specInput);
   const spec = specInput;
+  const that = Erizo.BaseStack(specInput);
+  const defaultSimulcastSpatialLayers = 2;
 
-  that.pcConfig = {
-    iceServers: [],
```
Comment (jcague) [inline erizo_controller/erizoClient/src/webrtc-stacks/ChromeStableStack.js]: nice! :+1:

File: erizo_controller/erizoClient/src/webrtc-stacks/ChromeStableStack.js [hunk 3]
```diff
@@ -99,286 +46,24 @@ Erizo.ChromeStableStack = (specInput) => {
...
       spatialLayerIdRtx = spatialLayersRtx[i];
-      result += addSpatialLayer(cname, msid, mslabel, label, spatialLayerId, spatialLayerIdRtx);
+      result += Erizo.SdpHelpers.addSpatialLayer(cname,
+        msid, mslabel, label, spatialLayerId, spatialLayerIdRtx);
     }
     result += 'a=x-google-flag:conference\r\n';
     return sdp.replace(matchGroup[0], result);
   };
 
-  const setMaxBW = (sdpInput) => {
```
Comment (jcague) [inline erizo_controller/erizoClient/src/webrtc-stacks/ChromeStableStack.js]: bye bye!

File: erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    .then(setLocalDescAux.bind(null, isSubscribe))
+    .catch(errorCallback.bind(null, 'Create Offer', undefined));
+  };
+
+  that.addStream = (stream) => {
+    that.peerConnection.addStream(stream);
+  };
+
+  that.processSignalingMessage = (msgInput) => {
+    // L.Logger.info("Process Signaling Message", msg);
```
Comment (jcague) [inline erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js]: Consider removing commented code.

Comment (lodoyun) [inline erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js]: 👍

**Pull request merged**